src/main/host/syscall/random.rs

```diff
@@ -0,0 +1,65 @@
+use crate::cshadow as c;
+use crate::host::context::{ThreadContext, ThreadContextObjs};
+use crate::host::syscall_types::{PluginPtr, SysCallArgs, SyscallResult, TypedPluginPtr};
+use rand::RngCore;
+
+use log::*;
+use nix::errno::Errno;
+
+use syscall_logger::log_syscall;
+
+#[log_syscall(/* rv */ libc::ssize_t, /* buf */ *const libc::c_void, /* count */ libc::size_t, /* flags */ libc::c_uint)]
+pub fn getrandom(ctx: &mut ThreadContext, args: &SysCallArgs) -> SyscallResult {
+    let buf_ptr: PluginPtr = args.get(0).into(); // char*
+    let count: libc::size_t = args.get(1).into();
+
+    // We ignore the flags arg, because we use the same random source for both
+    // random and urandom, and it never blocks anyway.
+
+    trace!("Trying to read {} random bytes.", count);
+
+    if buf_ptr.is_null() {
+        debug!("Invalid buffer.");
+        return Err(Errno::EFAULT.into());
+    }
+
+    // Get a native-process mem buffer where we can copy the random bytes.
+    let memory_manager = ctx.process.memory_mut();
+    let plugin_src = TypedPluginPtr::new::<u8>(PluginPtr::from(buf_ptr), count);
+    let memory_ref = memory_manager.memory_ref_mut_uninit(plugin_src);
+
+    match memory_ref {
```

Comment:
Stylistic nit - this could be unnested to something like:

```
let memory_ref = match memory_manager.memory_ref_mut_uninit(plugin_src) {
    Ok(m) => m,
    Err(e) => {
           warn!("Failed to get memory ref: {:?}", e);
           return Err(Errno::EFAULT.into())
    },
};

ctx.host.random().fill_bytes(&mut memory_ref);
match memory_ref.flush() {
...

```